vspace/src/main/java/edu/asu/diging/vspace/core/services/impl/ContentBlockManager.java

```diff
@@ -253,17 +286,42 @@ public IChoiceBlock getChoiceBlock(String choiceBlockId) {
     /*
      * (non-Javadoc)
      * 
-     * @see edu.asu.diging.vspace.core.services.impl.IChoiceBlock#createTextBlock(java.
+     * @see
+     * edu.asu.diging.vspace.core.services.impl.IChoiceBlock#createTextBlock(java.
      * lang.String, java.lang.String, java.lang.Integer)
      */
     @Override
-    public IChoiceBlock createChoiceBlock(String slideId, List<String> selectedChoices, Integer contentOrder, boolean showsAll) {
+    public IChoiceBlock createChoiceBlock(String slideId, List<String> selectedChoices, Integer contentOrder,
+            boolean showsAll) {
         List<IChoice> choices = new ArrayList<IChoice>();
-        if(!showsAll) {
-            choices = selectedChoices.stream().map(choice -> slideManager.getChoice(choice)).collect(Collectors.toList());
+        if (!showsAll) {
+            choices = selectedChoices.stream().map(choice -> slideManager.getChoice(choice))
+                    .collect(Collectors.toList());
         }
-        IChoiceBlock choiceBlock = choiceBlockFactory.createChoiceBlock(slideManager.getSlide(slideId), contentOrder, choices, showsAll);
-        return choiceBlockRepo.save((ChoiceBlock)choiceBlock);
+        IChoiceBlock choiceBlock = choiceBlockFactory.createChoiceBlock(slideManager.getSlide(slideId), contentOrder,
+                choices, showsAll);
+        return choiceBlockRepo.save((ChoiceBlock) choiceBlock);
     }
 
+    /**
+     * Retrieving the maximum content order for a slide
+     */
+    @Override
+    public Integer findMaxContentOrder(String slideId) {
+        return contentBlockRepository.findMaxContentOrder(slideId);
+    }
+
+    /**
+     * decreasing content order by 1 of the slide's block which are after the
+     * specified contentOrder
+     */
+    private void updateContentOrder(Integer contentOrder) {
+        if (contentOrder != null) {
+            List<ContentBlock> contentBlockList = contentBlockRepository.findByContentOrderGreaterThan(contentOrder);
+            for (ContentBlock eachContentBlock : contentBlockList) {
+                eachContentBlock.setContentOrder(eachContentBlock.getContentOrder() - 1);
+            }
+            contentBlockRepository.saveAll(contentBlockList);
+        }
```

Comment:
doesn't thism method update the content order for all content blocks no matter to which slide they belong? Not just the content blocks of a specific slide? that seems wrong.

Reply:
Yes. Slideid got missed.Updated the code.